syntax = "proto3";
option go_package = "./;trailer";
option java_multiple_files = false;
option java_package = "trailer";
option java_outer_classname = "trailer";

package trailer;

service trailer {
  // 初始化, 主要是为了传配置进去
  rpc Init (Config) returns (Response) {}
  // 启动
  rpc Start (Request) returns (Response) {}
  // 获取状态
  rpc Status (Request) returns (Response) {}
  // 服务调用
  rpc Service (ServiceRequest) returns (ServiceResponse) {}
  // 数据模型
  rpc Schema (SchemaRequest) returns (SchemaResponse) {}
  // 停止
  rpc Stop (Request) returns (Response) {}
}

message Config {
    map<string, string> kv = 1;
}
message Request {

}
message Response {
    int32 code = 1;
    string message = 2;
}
message ServiceRequest{
    bytes cmd = 1;
    bytes args = 2;
}
message ServiceResponse{
    int32 code = 1;
    bytes data = 2;
}
message StatusResponse {
    enum Status {
        RUNNING = 0;
        ERROR = 1;
        STOPED = 2;
    }
    Status status = 1;
    string message = 2;
}

message SchemaRequest{
}
message Column{
    bytes name = 1;
    bytes type = 2;
    bytes description = 3;
}
message SchemaResponse{
    int32 code = 1;
    bytes message = 2;
    repeated Column columns = 3;
}